### PR TITLE
crypto: Use sui read/write keypair logic in narwhal

### PR DESCRIPTION
**Pull request by joyqvq**
https://github.com/MystenLabs/sui/issues/5195 (Reopened since the benchmark are still using narwhal commands)

Test using `fab local` (all changes are mirrored in full_demo/remote/local) @arun-koshy lmk if there is any CI I should run to ensure correctness. 
```
fab local 
Starting local benchmark
Setting up testbed...
About to run ['cargo', 'build', '--quiet', '--release', '--features', 'benchmark']...
Running benchmark (60 sec)...
Parsing logs...
WARN: Clients missed their target rate 288 time(s)

-----------------------------------------
 SUMMARY:
-----------------------------------------
 + CONFIG:
 Faults: 0 node(s)
 Committee size: 4 node(s)
 Worker(s) per node: 1 worker(s)
 Collocate primary and workers: True
 Input rate: 50,000 tx/s
 Transaction size: 512 B
 Execution time: 57 s

 Header number of batches threshold: 32 digests
 Header maximum number of batches: 1,000 digests
 Max header delay: 2,000 ms
 GC depth: 50 round(s)
 Sync retry delay: 10,000 ms
 Sync retry nodes: 3 node(s)
 batch size: 500,000 B
 Max batch delay: 200 ms
 Max concurrent requests: 500,000

 + RESULTS:
 Batch creation avg latency: 74 ms
 Header creation avg latency: 2,168 ms
 	Batch to header avg latency: 1,392 ms
 Header to certificate avg latency: 8 ms
 	Request vote outbound avg latency: 4 ms
 Certificate commit avg latency: 2,930 ms

 Consensus TPS: 46,317 tx/s
 Consensus BPS: 23,714,522 B/s
 Consensus latency: 3,037 ms

 End-to-end TPS: 45,509 tx/s
 End-to-end BPS: 23,300,412 B/s
 End-to-end latency: 4,483 ms
-----------------------------------------
```

### CODE REVIEW

File: narwhal/benchmark/benchmark/config.py [hunk 1]
```diff
@@ -9,20 +9,6 @@
...
-    def __init__(self, name, secret):
-        self.name = name
-        self.secret = secret
-
-    @classmethod
-    def from_file(cls, filename):
-        assert isinstance(filename, str)
-        with open(filename, 'r') as f:
-            data = load(f)
-        return cls(data['name'], data['secret'])
```
Comment (joyqvq) [inline narwhal/benchmark/benchmark/config.py:23]: deprecate this old format

File: narwhal/benchmark/benchmark/commands.py [hunk 2]
```diff
@@ -33,11 +33,16 @@ def generate_key(filename):
         assert isinstance(filename, str)
         return f'./narwhal-node generate_keys --filename {filename}'
 
+    @staticmethod
+    def get_pk(filename):
```
Comment (arun-koshy) [inline narwhal/benchmark/benchmark/commands.py:37]: nit: changing this to `get_pub_key` might help clear up any confusion between pk meaning private or public

**Comment by joyqvq** [comment]
in narwhal we have `~/narwhal/crypto/src/lib.rs`
```
pub type PublicKey = bls12381::min_sig::BLS12381PublicKey;
pub type Signature = bls12381::min_sig::BLS12381Signature;
pub type AggregateSignature = bls12381::min_sig::BLS12381AggregateSignature;
pub type PrivateKey = bls12381::min_sig::BLS12381PrivateKey;
pub type KeyPair = bls12381::min_sig::BLS12381KeyPair;

pub type NetworkPublicKey = ed25519::Ed25519PublicKey;
pub type NetworkKeyPair = ed25519::Ed25519KeyPair;
```
in sui types `~/sui/crates/sui-types/src/crypto.rs`: 
```
// Authority Objects
pub type AuthorityKeyPair = BLS12381KeyPair;
pub type AuthorityPublicKey = BLS12381PublicKey;
pub type AuthorityPrivateKey = BLS12381PrivateKey;
pub type AuthoritySignature = BLS12381Signature;
pub type AggregateAuthoritySignature = BLS12381AggregateSignature;
pub type NetworkKeyPair = Ed25519KeyPair;
pub type NetworkPublicKey = Ed25519PublicKey;
pub type NetworkPrivateKey = Ed25519PrivateKey;
```
any benefits to keep two type alias since narwhal and sui are combined? imo we should just use sui-types as the only source of truth cc @kchalkias

File: narwhal/node/src/main.rs [hunk 3]
```diff
@@ -42,13 +47,18 @@ async fn main() -> Result<(), eyre::Report> {
...
-                .about("Print a fresh key pair to file")
-                .args_from_usage("--filename=<FILE> 'The file where to print the new key pair'"),
+                .about("Save an encoded bls12381 keypair (Base64 encoded `privkey`) to file")
+                .args_from_usage("--filename=<FILE> 'The file where to save the encoded authority key pair'"),
         )
         .subcommand(
             SubCommand::with_name("generate_network_keys")
-                .about("Print a fresh network key pair (ed25519) to file")
-                .args_from_usage("--filename=<FILE> 'The file where to print the new network key pair'"),
+            .about("Save an encoded ed25519 network keypair (Base64 encoede `flag || privkey`) to file")
```
Comment (arun-koshy) [inline narwhal/node/src/main.rs:55]: nit: 'encoded'

**Review by arun-koshy — APPROVED** [review]
Looks great! Thanks for making these changes 🚀

**Comment by joyqvq** [comment]
cc @arun-koshy, one more q, do we still use `~/sui/narwhal/Docker/validators` anywhere?

if possible, we probably should remove any of these hardcoded value. 

from what i see, `~/sui/narwhal/Docker/gen.validators.sh` and `gen.validators` and `gen.workers` should generate all these using the new encoding without pubkey.

**Comment by arun-koshy** [comment]
> lmk if there is any CI I should run to ensure correctness.

`fab local` should cover you

> do we still use ~/sui/narwhal/Docker/validators anywhere?

AFAIK our narwhal partner (Sommelier) are the only customers that might be using it, @akichidis can confirm.

**Comment by joyqvq** [comment]
landing this for now to unblock benchmarking, will follow up PR on 1) possibly remove all `~/sui/narwhal/Docker/validators` 2) collapse type alias from narwhal to sui

**Comment by arun-koshy** [comment]
Not sure if integrating parts of Sui into narwhal is something we are still trying to limit as narwhal was meant to be a separate entity in the long run. But in the short term maybe its okay? Will add @akichidis to the reviewers to make sure, as he is working the closest with our partners

**Pull request merged**